File: frontend/src/types/admin-users.ts
/**
 * Phase-80-Typen für die Admin-User-Übersicht und den User-Detail-Drawer.
 * Spiegelt die json-Tags aus backend/internal/models/admin_users.go 1:1.
 *
 * Basistypen (AppUser, CurrentUserData) werden aus auth.ts importiert,
 * nicht dupliziert.
 */

// ---------------------------------------------------------------------------
// Listen-Endpunkt
// ---------------------------------------------------------------------------

/** Eine Zeile in der paginierten Admin-User-Übersichtsliste. */
export interface AdminUserListItem {
  id: number
  email: string
  display_name: string
  status: 'pending' | 'active' | 'disabled'
  global_roles: string[]

  // Member-Profil-Anker (via member_claims, verified)
  member_profile_id: number | null
  member_profile_name: string | null

  // D-05-Aggregat-Counts
  group_membership_count: number
  leader_context_count: number
  open_claims_count: number
  open_contributions_count: number
  total_contributions_count: number
  media_upload_count: number
  release_scope_count: number
  conflict_count: number
  last_activity_at: string | null
}

/** Filter- und Paginierungsparameter für die Admin-User-Liste. */
export interface AdminUserListParams {
  q?: string
  status?: string
  global_role?: string
  has_conflicts?: boolean
  sort?: string
  limit?: number
  offset?: number
}

/** Response-Envelope der paginierten Admin-User-Liste. */
export interface AdminUserListResponse {
  data: AdminUserListItem[]
  meta: {
    total: number
    limit: number
    offset: number
  }
}

// ---------------------------------------------------------------------------
// Übersicht-Tab
// ---------------------------------------------------------------------------

/** Ein einzelner Konflikt-Eintrag im Übersicht-Tab. */
export interface AdminConflictDetail {
  type: string
  message: string
}

/** Response des Übersicht-Tab-Endpunkts. */
export interface AdminUserOverviewResponse {
  id: number
  email: string
  display_name: string
  status: 'pending' | 'active' | 'disabled'
  global_roles: string[]

  // Zusammenfassung-Counts
  group_membership_count: number
  leader_context_count: number
  open_claims_count: number
  open_contributions_count: number
  total_contributions_count: number
  media_upload_count: number
  release_scope_count: number

  // Conflict-Aufschlüsselung (D-19)
  conflict_details: AdminConflictDetail[]

  // Zeitstempel
  last_login_at: string | null
  created_at: string
  updated_at: string
}

// ---------------------------------------------------------------------------
// Rollen-Tab
// ---------------------------------------------------------------------------

/** Response des Rollen-Tab-Endpunkts. */
export interface AdminUserGlobalRolesResponse {
  roles: string[]
  assignable_roles: string[]
}

// ---------------------------------------------------------------------------
// Claims-Tab
// ---------------------------------------------------------------------------

export interface AdminMemberProfileSummary {
  member_id: number
  fansub_name: string
  profile_status: string
  avatar_url: string | null
}

export interface AdminClaimSummary {
  claim_id: number
  claim_type: string
  claim_status: string
  member_id: number
  fansub_name: string
  created_at: string
  resolved_at: string | null
}

/** Response des Claims-Tab-Endpunkts. */
export interface AdminUserMemberClaimsResponse {
  member_profile: AdminMemberProfileSummary | null
  claims: AdminClaimSummary[]
}

// ---------------------------------------------------------------------------
// Gruppenmitgliedschaften-Tab
// ---------------------------------------------------------------------------

export interface AdminGroupMembershipSummary {
  fansub_group_id: number
  fansub_group_name: string
  member_status: string
  roles: string[]
  joined_at: string
}

/** Response des Gruppenmitgliedschaften-Tab-Endpunkts. */
export interface AdminUserGroupMembershipsResponse {
  memberships: AdminGroupMembershipSummary[]
}

// ---------------------------------------------------------------------------
// Gruppenrechte-Tab (read-only, D-03)
// ---------------------------------------------------------------------------

export interface AdminGroupRightsSummary {
  fansub_group_id: number
  fansub_group_name: string
  granted_roles: string[]
  can_view_members: boolean
  can_edit_content: boolean
}

/** Response des Gruppenrechte-Tab-Endpunkts (read-only). */
export interface AdminUserGroupRightsResponse {
  group_rights: AdminGroupRightsSummary[]
}

// ---------------------------------------------------------------------------
// Contributions-Tab (Phase-83-aware, D-13)
// ---------------------------------------------------------------------------

/** Eine einzelne Contribution eines Users. */
export interface AdminContributionItem {
  contribution_id: number
  fansub_group_id: number
  fansub_group_name: string
  anime_id: number
  anime_title: string
  release_version_id: number | null
  contribution_type: 'project_default' | 'release_override'
  dispute_state: string
  role_codes: string[]
  // D-29: die echte fachliche Version (release_versions.version) und die
  // Episodennummer für die Anzeige — release_version_id bleibt intern
  // und wird nie mehr roh im UI gerendert. episode_number ist string
  // (episodes.episode_number ist in der Datenbank TEXT), nicht number.
  release_version_label: string | null
  episode_number: string | null
}

/** Response des Contributions-Tab-Endpunkts (vier Gruppen, D-13). */
export interface AdminUserContributionsResponse {
  project_defaults: AdminContributionItem[]
  release_overrides: AdminContributionItem[]
  open_disputes: AdminContributionItem[]
  legacy_historical: AdminContributionItem[]
}

// ---------------------------------------------------------------------------
// Medien-Tab
// ---------------------------------------------------------------------------

export interface AdminMediaItemSummary {
  media_asset_id: number
  media_type: string
  original_filename: string
  public_url: string
  file_size_bytes: number
  uploaded_at: string
  owner_context: string
}

/** Response des Medien-Tab-Endpunkts. */
export interface AdminUserMediaResponse {
  media_items: AdminMediaItemSummary[]
}

// ---------------------------------------------------------------------------
// Audit-Tab
// ---------------------------------------------------------------------------

export interface AdminAuditEntry {
  event_id: number
  event_type: string
  target_type: string
  target_id: number | null
  action: string
  outcome: string
  occurred_at: string
}

/** Response des Audit-Tab-Endpunkts. */
export interface AdminUserAuditResponse {
  entries: AdminAuditEntry[]
}

// ---------------------------------------------------------------------------
// Zentrale Claims-Arbeitsqueue (D-23, Plan 138-05)
// ---------------------------------------------------------------------------

/** Eine Zeile in der gruppenuebergreifenden Claims-Arbeitsqueue. */
export interface AdminClaimListRow {
  claim_id: number
  app_user_id: number
  app_user_email: string
  app_user_display_name: string
  member_id: number
  member_nickname: string
  claim_status: 'pending' | 'verified' | 'rejected'
  claim_type: string
  fansub_group_id: number
  fansub_group_name: string
  note: string
  created_at: string
  verified_at: string | null
}

/** Filter- und Paginierungsparameter für GET /admin/claims. */
export interface AdminClaimsListParams {
  status?: string
  claim_type?: string
  fansub_group_id?: number
  app_user_id?: number
  from?: string
  to?: string
  limit?: number
  offset?: number
}

/** Response-Envelope der zentralen Claims-Arbeitsqueue. */
export interface AdminClaimsListResponse {
  data: AdminClaimListRow[]
  meta: {
    total: number
    limit: number
    offset: number
  }
}

// ---------------------------------------------------------------------------
// Zentrale Aenderungen/Audit-Arbeitsqueue (D-25/D-28, Plan 138-05)
// ---------------------------------------------------------------------------

/** Eine Zeile in der gruppenuebergreifenden Aenderungen-Arbeitsqueue. */
export interface AdminChangeEntry {
  event_id: number
  event_type: string
  target_type: string
  target_id: number | null
  action: string
  outcome: string
  occurred_at: string
  actor_app_user_id: number | null
  scope_type: string
  scope_id: number | null
  payload: Record<string, unknown> | null
  actor_display_name: string | null
  target_display_name: string | null
}

/** Filter- und Paginierungsparameter für GET /admin/changes. */
export interface AdminChangesListParams {
  benutzer?: number
  akteur?: number
  gruppe?: number
  target_type?: string
  from?: string
  to?: string
  limit?: number
  offset?: number
}

/** Response-Envelope der zentralen Aenderungen-Arbeitsqueue. */
export interface AdminChangesListResponse {
  data: AdminChangeEntry[]
  meta: {
    total: number
    limit: number
    offset: number
  }
}

// ---------------------------------------------------------------------------
// Phase-139: skalierbare User-Admin-Projektionen (Beitraege/Medien/Rechte)
//
// Diese Typen sind additiv. Sie ersetzen NICHT AdminUserContributionsResponse/
// AdminUserMediaResponse/AdminContributionItem/AdminMediaItemSummary, deren
// Envelope-Form weiterhin von anderen Konsumenten genutzt wird — spiegeln
// 1:1 die json-Tags aus backend/internal/models/admin_users.go (Plan 139-01).
// ---------------------------------------------------------------------------

/** Geteilte Paginierungs-Meta-Huelle fuer neue Phase-139-Listen-Endpunkte. */
export interface AdminListMeta {
  total: number
  limit: number
  offset: number
}

/** Generisches {id,name}-Paar fuer filter_options-Eintraege. */
export interface AdminFilterOption {
  id: number
  name: string
}

/** Die immer sichtbare Projektstandard-Zeile (D03). */
export interface AdminContributionStandardSummary {
  role_codes: string[]
  contributor_labels: string[]
}

/**
 * Ein zusammengefasster Bereich oder eine einzelne Abweichung (D04/D05/D06).
 * is_deviation wird ausschliesslich durch einen echten semantischen Diff
 * gesetzt, niemals durch reine Anwesenheit von release_version_id.
 */
export interface AdminContributionRangeEntry {
  from_label: string
  to_label: string
  is_deviation: boolean
  deviation_detail: string | null
  role_codes: string[]
}

/** Ein Seiten-Element: ein Anime+Projekt-Block (D08). */
export interface AdminContributionProjectBlock {
  anime_id: number
  anime_title: string
  fansub_group_id: number
  fansub_group_name: string
  project_standard: AdminContributionStandardSummary
  range_entries: AdminContributionRangeEntry[]
}

/**
 * Auf die Beitragshistorie des jeweils einen Users beschraenkt (UI-SPEC
 * Filter Data Contract) — niemals ein zweiter unbegrenzter Lookup-Endpunkt.
 */
export interface AdminContributionFilterOptions {
  animes: AdminFilterOption[]
  groups: AdminFilterOption[]
}

/** Die neue paginierte Response-Huelle fuer den Contributions-Tab. */
export interface AdminUserContributionsPage {
  data: AdminContributionProjectBlock[]
  meta: AdminListMeta
  filter_options: AdminContributionFilterOptions
}

/**
 * Ein einzelnes Media-Item innerhalb eines Release-/Episoden-Blocks. Bewusst
 * OHNE owner_context-Feld und ohne Scope-/Berechtigungs-Flag (D19) — dies
 * ist ein neuer Typ, keine Mutation des bestehenden AdminMediaItemSummary.
 */
export interface AdminMediaItem {
  media_asset_id: number
  media_type: string
  original_filename: string
  public_url: string
  file_size_bytes: number
  uploaded_at: string
}

/** Ein Seiten-Element: ein Release-/Episoden-Block (D12). */
export interface AdminMediaReleaseBlock {
  anime_id: number
  anime_title: string
  fansub_group_id: number
  fansub_group_name: string
  release_version_id: number
  release_version_label: string
  episode_number: string | null
  items: AdminMediaItem[]
}

/** Die verfuegbaren Filter fuer den Medien-Tab. */
export interface AdminMediaFilterOptions {
  animes: AdminFilterOption[]
  groups: AdminFilterOption[]
  releases_or_episodes: AdminFilterOption[]
  media_types: string[]
}

/** Die neue paginierte Response-Huelle fuer den Medien-Tab. */
export interface AdminUserMediaPage {
  data: AdminMediaReleaseBlock[]
  meta: AdminListMeta
  filter_options: AdminMediaFilterOptions
}

/** F-01-Kompaktzusammenfassung einer einzelnen Faehigkeit. */
export interface AdminHeadlineCapabilityState {
  action_code: string
  label: string
  allowed: boolean
}

/** F-01-Seiten-Element: eine Gruppe mit kompakter Rechte-Zusammenfassung. */
export interface AdminUserGroupRightsSummaryItem {
  fansub_group_id: number
  fansub_group_name: string
  role_label: string
  headline_states: AdminHeadlineCapabilityState[]
  has_deviation: boolean
  open_claims_count: number
}

/** Die neue paginierte, gebuendelte F-01-Response-Huelle fuer die Rechte-Uebersicht. */
export interface AdminUserRightsSummaryPage {
  data: AdminUserGroupRightsSummaryItem[]
  meta: AdminListMeta
}

/** Filter- und Paginierungsparameter fuer GET /admin/users/:userId/contributions. */
export interface AdminUserContributionsParams {
  anime_id?: number
  fansub_group_id?: number
  role_code?: string
  only_deviations?: boolean
  from?: string
  to?: string
  limit?: number
  offset?: number
}

/** Filter- und Paginierungsparameter fuer GET /admin/users/:userId/media. */
export interface AdminUserMediaParams {
  anime_id?: number
  fansub_group_id?: number
  release_version_id?: number
  media_type?: string
  from?: string
  to?: string
  limit?: number
  offset?: number
}

File: backend/internal/models/admin_users.go
package models

import "time"

// Conflict-Typ-Konstanten für Phase-80-Conflict-Detection (D-17/D-18).
// Handler validiert Type-Wert vor DB-Query (Phase 80-03, T-80-01-01).
const (
	AdminConflictTypeOpenClaim                      = "open_claim_with_profile"
	AdminConflictTypeMemberWithoutRole              = "member_without_role"
	AdminConflictTypeMediaWithoutScope              = "media_without_scope"
	AdminConflictTypeOpenDispute                    = "open_dispute"
	AdminConflictTypeInvalidReleaseOverride         = "invalid_release_override"
	AdminConflictTypeOverrideContradiction          = "override_contradiction"
	AdminConflictTypeMediaWithoutContributionRights = "media_without_contribution_rights"
)

// AdminUserListItem ist eine Zeile in der paginierten Admin-User-Übersichtsliste.
// Enthält AppUser-Basisfelder plus alle D-05-Aggregat-Counts.
type AdminUserListItem struct {
	ID          int64    `json:"id"`
	Email       string   `json:"email"`
	DisplayName string   `json:"display_name"`
	Status      string   `json:"status"`
	GlobalRoles []string `json:"global_roles"`

	// Member-Profil-Anker (via member_claims, verified)
	MemberProfileID   *int64  `json:"member_profile_id"`
	MemberProfileName *string `json:"member_profile_name"`

	// D-05-Aggregat-Counts
	GroupMembershipCount      int    `json:"group_membership_count"`
	LeaderContextCount        int    `json:"leader_context_count"`
	OpenClaimsCount           int    `json:"open_claims_count"`
	OpenContributionsCount    int    `json:"open_contributions_count"`
	TotalContributionsCount   int    `json:"total_contributions_count"`
	MediaUploadCount          int    `json:"media_upload_count"`
	ReleaseScopeCount         int    `json:"release_scope_count"`
	ConflictCount             int    `json:"conflict_count"`
	LastActivityAt            *string `json:"last_activity_at"`
}

// AdminUserListParams enthält Filter- und Paginierungsparameter für die User-Liste.
type AdminUserListParams struct {
	Q            string `json:"q"`
	Status       string `json:"status"`
	GlobalRole   string `json:"global_role"`
	HasConflicts bool   `json:"has_conflicts"`
	Sort         string `json:"sort"`
	Limit        int    `json:"limit"`
	Offset       int    `json:"offset"`
}

// AdminUserListResult ist das Response-DTO der paginierten User-Liste.
type AdminUserListResult struct {
	Data []AdminUserListItem `json:"data"`
	Meta struct {
		Total  int `json:"total"`
		Limit  int `json:"limit"`
		Offset int `json:"offset"`
	} `json:"meta"`
}

// AdminConflict beschreibt einen einzelnen Konflikt im Übersicht-Tab des User-Detail-Drawers.
type AdminConflict struct {
	Type    string `json:"type"`
	Message string `json:"message"`
}

// AdminUserOverview ist das Haupt-DTO für den Übersicht-Tab des User-Detail-Drawers.
type AdminUserOverview struct {
	ID          int64    `json:"id"`
	Email       string   `json:"email"`
	DisplayName string   `json:"display_name"`
	Status      string   `json:"status"`
	GlobalRoles []string `json:"global_roles"`

	// Zusammenfassung-Counts (D-05)
	GroupMembershipCount    int `json:"group_membership_count"`
	LeaderContextCount      int `json:"leader_context_count"`
	OpenClaimsCount         int `json:"open_claims_count"`
	OpenContributionsCount  int `json:"open_contributions_count"`
	TotalContributionsCount int `json:"total_contributions_count"`
	MediaUploadCount        int `json:"media_upload_count"`
	ReleaseScopeCount       int `json:"release_scope_count"`

	// Conflict-Aufschlüsselung (D-19)
	ConflictDetails []AdminConflict `json:"conflict_details"`

	// Zeitstempel
	LastLoginAt *time.Time `json:"last_login_at,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

// AdminUserGlobalRolesResult ist das DTO für den Rollen-Tab.
type AdminUserGlobalRolesResult struct {
	Roles           []string `json:"roles"`
	AssignableRoles []string `json:"assignable_roles"`
}

// AdminMemberProfileSummary fasst das verlinktes Member-Profil zusammen.
type AdminMemberProfileSummary struct {
	MemberID      int64   `json:"member_id"`
	FansubName    string  `json:"fansub_name"`
	ProfileStatus string  `json:"profile_status"`
	AvatarURL     *string `json:"avatar_url,omitempty"`
}

// AdminClaimSummary beschreibt einen einzelnen Claim/eine Einladung.
type AdminClaimSummary struct {
	ClaimID       int64   `json:"claim_id"`
	ClaimType     string  `json:"claim_type"`
	ClaimStatus   string  `json:"claim_status"`
	MemberID      int64   `json:"member_id"`
	FansubName    string  `json:"fansub_name"`
	CreatedAt     string  `json:"created_at"`
	ResolvedAt    *string `json:"resolved_at,omitempty"`
}

// AdminUserMemberClaimsResult ist das DTO für den Claims-Tab.
type AdminUserMemberClaimsResult struct {
	MemberProfile *AdminMemberProfileSummary `json:"member_profile"`
	Claims        []AdminClaimSummary        `json:"claims"`
}

// AdminGroupMembershipSummary beschreibt eine einzelne Grupenmitgliedschaft.
type AdminGroupMembershipSummary struct {
	FansubGroupID   int64    `json:"fansub_group_id"`
	FansubGroupName string   `json:"fansub_group_name"`
	MemberStatus    string   `json:"member_status"`
	Roles           []string `json:"roles"`
	JoinedAt        string   `json:"joined_at"`
}

// AdminUserGroupMembershipsResult ist das DTO für den Gruppenmitgliedschaften-Tab.
type AdminUserGroupMembershipsResult struct {
	Memberships []AdminGroupMembershipSummary `json:"memberships"`
}

// AdminGroupRightsSummary beschreibt die scoped Rechte eines Users in einer Gruppe (read-only, D-03).
type AdminGroupRightsSummary struct {
	FansubGroupID   int64    `json:"fansub_group_id"`
	FansubGroupName string   `json:"fansub_group_name"`
	GrantedRoles    []string `json:"granted_roles"`
	CanViewMembers  bool     `json:"can_view_members"`
	CanEditContent  bool     `json:"can_edit_content"`
}

// AdminUserGroupRightsResult ist das DTO für den Gruppenrechte-Tab (read-only).
type AdminUserGroupRightsResult struct {
	GroupRights []AdminGroupRightsSummary `json:"group_rights"`
}

// AdminContributionItem beschreibt eine einzelne Contribution eines Users.
// Orientiert sich an Phase-83-Auflösungslogik (member_id-Anker, D-12/D-13).
type AdminContributionItem struct {
	ContributionID   int64    `json:"contribution_id"`
	FansubGroupID    int64    `json:"fansub_group_id"`
	FansubGroupName  string   `json:"fansub_group_name"`
	AnimeID          int64    `json:"anime_id"`
	AnimeTitle       string   `json:"anime_title"`
	ReleaseVersionID *int64   `json:"release_version_id"`
	ContributionType string   `json:"contribution_type"` // "project_default" | "release_override"
	DisputeState     string   `json:"dispute_state"`
	RoleCodes        []string `json:"role_codes"`

	// D-29: die echte fachliche Version (release_versions.version) und die
	// Episodennummer für die Anzeige — release_version_id bleibt intern
	// (Foreign Key), wird aber nie mehr roh im UI gerendert. Beide nullable,
	// weil ReleaseVersionID selbst bei Projekt-Standard-Beiträgen NULL ist.
	// Beide Felder sind bewusst *string (nicht *int): der Episodenspalten-Typ
	// ist in der Datenbank TEXT (Migration 0002), siehe die gleiche Konvention
	// in GroupReleaseVersionOption (anime_contributions_release_lookup_repository.go).
	ReleaseVersionLabel *string `json:"release_version_label"`
	EpisodeNumber       *string `json:"episode_number"`
}

// AdminUserContributionsResult ist das DTO für den Contributions-Tab (D-13: vier Gruppen).
type AdminUserContributionsResult struct {
	ProjectDefaults   []AdminContributionItem `json:"project_defaults"`
	ReleaseOverrides  []AdminContributionItem `json:"release_overrides"`
	OpenDisputes      []AdminContributionItem `json:"open_disputes"`
	LegacyHistorical  []AdminContributionItem `json:"legacy_historical"`
}

// AdminMediaItemSummary beschreibt ein einzelnes Media-Item eines Users.
type AdminMediaItemSummary struct {
	MediaAssetID     int64   `json:"media_asset_id"`
	MediaType        string  `json:"media_type"`
	OriginalFilename string  `json:"original_filename"`
	PublicURL        string  `json:"public_url"`
	FileSizeBytes    int64   `json:"file_size_bytes"`
	UploadedAt       string  `json:"uploaded_at"`
	OwnerContext     string  `json:"owner_context"` // z. B. "release_version:42"
}

// AdminUserMediaResult ist das DTO für den Medien-Tab.
type AdminUserMediaResult struct {
	MediaItems []AdminMediaItemSummary `json:"media_items"`
}

// AdminAuditEntry beschreibt einen einzelnen Audit-Log-Eintrag.
type AdminAuditEntry struct {
	EventID    int64   `json:"event_id"`
	EventType  string  `json:"event_type"`
	TargetType string  `json:"target_type"`
	TargetID   *int64  `json:"target_id,omitempty"`
	Action     string  `json:"action"`
	Outcome    string  `json:"outcome"`
	OccurredAt string  `json:"occurred_at"`
}

// AdminUserAuditResult ist das DTO für den Audit-Tab.
type AdminUserAuditResult struct {
	Entries []AdminAuditEntry `json:"entries"`
}

// Phase-139: neue gruppierte/paginierte DTOs für Contributions-, Media- und
// F-01-Rechte-Übersicht (Scalable User-Admin Projections).
//
// Diese DTOs sind additiv. Sie ersetzen NICHT AdminUserContributionsResult/
// AdminUserMediaResult/AdminContributionItem/AdminMediaItemSummary, deren
// ProjectDefaults/ReleaseOverrides/OpenDisputes/LegacyHistorical-Form und
// TotalContributionsCount-Feld weiterhin live produktive Oberflächen für
// andere Konsumenten sind (AdminUserOverview, AdminUserListItem) —
// AdminUserContributionsPage.Meta.Total zählt gefilterte PROJEKT-BLÖCKE
// (D09), eine andere Zahl mit anderer Bedeutung als TotalContributionsCount's
// rohe Zeilenzahl.

// AdminListMeta ist die geteilte Paginierungs-Meta-Hülle für neue Phase-139-
// Listen-Endpunkte (Data []X + Meta{Total,Limit,Offset}, mirrors
// AdminUserListResult's anonymes Meta-Struct, hier aber benannt, da drei neue
// DTOs sie wiederverwenden).
type AdminListMeta struct {
	Total  int `json:"total"`
	Limit  int `json:"limit"`
	Offset int `json:"offset"`
}

// AdminFilterOption ist ein generisches {id,name}-Paar für filter_options-
// Einträge (UI-SPEC Filter Data Contract).
type AdminFilterOption struct {
	ID   int64  `json:"id"`
	Name string `json:"name"`
}

// AdminContributionStandardSummary ist die immer sichtbare Projektstandard-
// Zeile (D03).
type AdminContributionStandardSummary struct {
	RoleCodes         []string `json:"role_codes"`
	ContributorLabels []string `json:"contributor_labels"`
}

// AdminContributionRangeEntry ist ein zusammengefasster Bereich oder eine
// einzelne Abweichung (D04/D05/D06). IsDeviation wird ausschließlich durch
// einen echten semantischen Diff gesetzt, niemals durch reine Anwesenheit
// von release_version_id.
type AdminContributionRangeEntry struct {
	FromLabel       string   `json:"from_label"`
	ToLabel         string   `json:"to_label"`
	IsDeviation     bool     `json:"is_deviation"`
	DeviationDetail *string  `json:"deviation_detail"`
	RoleCodes       []string `json:"role_codes"`
}

// AdminContributionProjectBlock ist ein Seiten-Element: ein Anime+Projekt-
// Block (D08).
type AdminContributionProjectBlock struct {
	AnimeID         int64                            `json:"anime_id"`
	AnimeTitle      string                           `json:"anime_title"`
	FansubGroupID   int64                            `json:"fansub_group_id"`
	FansubGroupName string                           `json:"fansub_group_name"`
	ProjectStandard AdminContributionStandardSummary `json:"project_standard"`
	RangeEntries    []AdminContributionRangeEntry    `json:"range_entries"`
}

// AdminContributionFilterOptions ist auf die Beitragshistorie des jeweils
// einen Users beschränkt (UI-SPEC Filter Data Contract) — niemals ein
// zweiter unbegrenzter Lookup-Endpunkt.
type AdminContributionFilterOptions struct {
	Animes []AdminFilterOption `json:"animes"`
	Groups []AdminFilterOption `json:"groups"`
}

// AdminUserContributionsPage ist die neue paginierte Response-Hülle für den
// Contributions-Tab.
type AdminUserContributionsPage struct {
	Data          []AdminContributionProjectBlock `json:"data"`
	Meta          AdminListMeta                    `json:"meta"`
	FilterOptions AdminContributionFilterOptions   `json:"filter_options"`
}

// AdminMediaItem ist ein einzelnes Media-Item innerhalb eines Release-/
// Episoden-Blocks. Bewusst OHNE OwnerContext-Feld und ohne Scope-/
// Berechtigungs-Flag (D19) — dies ist ein neuer Typ, keine Mutation des
// bestehenden AdminMediaItemSummary, der sein OwnerContext-Feld zur
// Rückwärtskompatibilität mit dem unveränderten alten flachen Endpunkt
// behält, bis Plan 139-04 dessen Aufrufer ersetzt.
type AdminMediaItem struct {
	MediaAssetID     int64  `json:"media_asset_id"`
	MediaType        string `json:"media_type"`
	OriginalFilename string `json:"original_filename"`
	PublicURL        string `json:"public_url"`
	FileSizeBytes    int64  `json:"file_size_bytes"`
	UploadedAt       string `json:"uploaded_at"`
}

// AdminMediaReleaseBlock ist ein Seiten-Element: ein Release-/Episoden-Block
// (D12).
type AdminMediaReleaseBlock struct {
	AnimeID             int64            `json:"anime_id"`
	AnimeTitle          string           `json:"anime_title"`
	FansubGroupID       int64            `json:"fansub_group_id"`
	FansubGroupName     string           `json:"fansub_group_name"`
	ReleaseVersionID    int64            `json:"release_version_id"`
	ReleaseVersionLabel string           `json:"release_version_label"`
	EpisodeNumber       *string          `json:"episode_number"`
	Items               []AdminMediaItem `json:"items"`
}

// AdminMediaFilterOptions beschreibt die verfügbaren Filter für den Medien-Tab.
type AdminMediaFilterOptions struct {
	Animes             []AdminFilterOption `json:"animes"`
	Groups             []AdminFilterOption `json:"groups"`
	ReleasesOrEpisodes []AdminFilterOption `json:"releases_or_episodes"`
	MediaTypes         []string            `json:"media_types"`
}

// AdminUserMediaPage ist die neue paginierte Response-Hülle für den Medien-Tab.
type AdminUserMediaPage struct {
	Data          []AdminMediaReleaseBlock `json:"data"`
	Meta          AdminListMeta            `json:"meta"`
	FilterOptions AdminMediaFilterOptions  `json:"filter_options"`
}

// AdminHeadlineCapabilityState ist die F-01-Kompaktzusammenfassung einer
// einzelnen Fähigkeit.
type AdminHeadlineCapabilityState struct {
	ActionCode string `json:"action_code"`
	Label      string `json:"label"`
	Allowed    bool   `json:"allowed"`
}

// AdminUserGroupRightsSummaryItem ist ein F-01-Seiten-Element: eine Gruppe
// mit kompakter Rechte-Zusammenfassung.
type AdminUserGroupRightsSummaryItem struct {
	FansubGroupID   int64                          `json:"fansub_group_id"`
	FansubGroupName string                         `json:"fansub_group_name"`
	RoleLabel       string                         `json:"role_label"`
	HeadlineStates  []AdminHeadlineCapabilityState `json:"headline_states"`
	HasDeviation    bool                           `json:"has_deviation"`
	OpenClaimsCount int                            `json:"open_claims_count"`
}

// AdminUserRightsSummaryPage ist die neue paginierte, gebündelte F-01-Response-
// Hülle für die Rechte-Übersicht.
type AdminUserRightsSummaryPage struct {
	Data []AdminUserGroupRightsSummaryItem `json:"data"`
	Meta AdminListMeta                     `json:"meta"`
}
